src/Akka.Streams.Kafka/Messages/CommittableOffsetBatch.cs

```diff
@@ -86,66 +85,82 @@ public ICommittableOffsetBatch Updated(ICommittable offset)
+            var key = partitionOffset.GroupTopicPartition;
+            var metadata = (newOffset is ICommittableOffsetMetadata withMetadata) ? withMetadata.Metadata : string.Empty;
 
-            var newOffsets = OffsetsAndMetadata.SetItem(partitionOffset.GroupTopicPartition, new OffsetAndMetadata(partitionOffset.Offset, metadata));
-            var committer = committableOffset is CommittableOffset c 
-                ? c.Committer 
-                : throw new ArgumentException($"Unknown committable offset, got {committableOffset.GetType().Name}, expected {nameof(committableOffset)}");
-            
-            
-            IImmutableDictionary<string, IInternalCommitter> newCommitters;
-            if (Committers.TryGetValue(partitionOffset.GroupId, out var groupCommitter))
-            {
-                if (!groupCommitter.Equals(committer))
-                {
-                    throw new ArgumentException($"CommittableOffset {committableOffset} committer for groupId {partitionOffset.GroupId} " +
-                                                $"must be same as the other with this groupId.");
-                }
+            var newOffsets = OffsetsAndMetadata.SetItem(key, new OffsetAndMetadata(partitionOffset.Offset, metadata));
 
-                newCommitters = Committers;
-            }
-            else
+            var newCommitter = newOffset switch
             {
-                newCommitters = Committers.SetItem(partitionOffset.GroupId, committer);
-            }
+                CommittableOffset c => c.Committer,
+                _ => throw new ArgumentOutOfRangeException(nameof(newOffset),
+                    $"Unknown committable offset, got {newOffset.GetType().Name}, expected {nameof(CommittableOffset)}")
+            };
+            
+            // the last KafkaAsyncConsumerCommitter for this GroupTopicPartition wins
+            var newCommitters = Committers.SetItem(key, newCommitter);
             
             return new CommittableOffsetBatch(newOffsets, newCommitters, BatchSize + 1);
         }
+        
+        internal KafkaAsyncConsumerCommitter CommitterFor(GroupTopicPartition groupTopicPartition)
+        {
+            if (Committers.TryGetValue(groupTopicPartition, out var committer))
+            {
+                return committer;
+            }
+            throw new ArgumentException($"Unknown committer for groupId {groupTopicPartition}");
+        }
+
+        internal CommittableOffsetBatch Filter(Predicate<GroupTopicPartition> p)
+        {
+            var newOffsets = OffsetsAndMetadata.Where(o => p(o.Key))
+                .ToImmutableDictionary(o => o.Key, o => o.Value);
+            var newCommiters =
+                Offsets.ToImmutableDictionary(c => c.GroupTopicPartition, v => CommitterFor(v.GroupTopicPartition));
+            return new CommittableOffsetBatch(newOffsets, newCommiters, BatchSize);
+        }
+        
+        public CommittableOffsetBatch TellCommit()
```

Comment:
These methods all produce the commit messages that we're going to send to the `KafkaConsumerActor`.